mesh_generation/generate_rof_weights.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2025 ACCESS-NRI and contributors. See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# =========================================================================================
+# Generate an remapping weights between two ESMF mesh files for remapping a runoff field
+# an unmasked mesh to a masked mesh without loosing any water volume. Each field on the
+# unmasked mesh is mapped to the nearest ocean cell in the resulting weights.
+#
+# To run:
+#   python generate_rof_weights.py --mesh_filename=<input_file> --weights_filename=<output_file>
+#
+# This script currently supports mesh files in the ESMF unstructed mesh format.
+#
+# There is not enough memory on the gadi login node to run this, its simplest to run in
+# a terminal through are.nci.org.au
+#
+# The run command and full github url of the current version of this script is added to the
+# metadata of the generated weights file. This is to uniquely identify the script and inputs used
+# to generate the mesh file. To produce weights files for sharing, ensure you are using a version
+# of this script which is committed and pushed to github. For mesh files intended for released
+# configurations, use the latest version checked in to the main branch of the github repository.
+#
+# Contact:
+#   Anton Steketee <anton.steketee@anu.edu.au>
+#
+# Dependencies:
+#   esmpy, xarray and scipy
+# =========================================================================================
+
+
+import xarray as xr
+import esmpy
+from scipy.spatial import KDTree
+from copy import copy
+
+from pathlib import Path
+import sys
+import os
+from datetime import datetime
+
+path_root = Path(__file__).parents[1]
+sys.path.append(str(path_root))
+from scripts_common import get_provenance_metadata, md5sum
+
+TEMP_WEIGHTS_F = "temp_weights.nc"
+COMP_ENCODING = {"complevel": 1, "compression": "zlib"}  # compression settings to use
+
+
+def drof_remapping_weights(mesh_filename, weights_filename, global_attrs=None):
+    # We need to generate remapping weights for use in the mediator, such that the overall volume of runoff is conserved and no run-off is mapped onto land cells. Inside the mediator, the grid doesn't change as we run the mediator with the ocean grid (the DROF component does the remapping from JRA grid to mediator grid). There we use the same _mesh_file for the input and output mesh, however this same routine would work for differeing input and output meshes
+
+    model_mesh = esmpy.Mesh(
+        filename=mesh_filename,
+        filetype=esmpy.FileFormat.ESMFMESH,
+    )
+
+    med_in_fld = esmpy.Field(model_mesh, meshloc=esmpy.MeshLoc.ELEMENT)
+
+    med_out_fld = esmpy.Field(model_mesh, meshloc=esmpy.MeshLoc.ELEMENT)
+
+    try:
+        os.remove(TEMP_WEIGHTS_F)  # rm old temp file
+    except:
+        None
```

Comment:
```suggestion
    except OSError:
        pass
```
https://stackoverflow.com/questions/10840533/most-pythonic-way-to-delete-a-file-which-may-not-exist